R/table-subscript.R

```diff
@@ -791,9 +791,11 @@ updateSpanIfNecessary <- function(y, x.attributes, evaluated.args) {
         span.df <- mapply(subscriptSpanDF, span.attribute, evaluated.args, SIMPLIFY = FALSE)
         span.df <- Filter(ncol, span.df)
         if (is.null(ncol(y)) && !is.null(span.df$columns)) {
+            nc <- ncol(span.df$columns)
             # When subscripting a row of a matrix results in a vector (i.e. drop = TRUE)
             # this is shown as a column vector even though transpose has not been called
-            span.df$rows <- span.df$columns
+            if (length(names(y)) == nrow(span.df$columns) && all(span.df$columns[,nc] == names(y)))
```

Comment:
In what situations will this be false? I assume that would be the case found by the bug, but I don't see spans in the input table.

Reply:
Looking at the tests, I guess it is based on which dimension was subscripted? Perhaps update the comment above to state this.

Reply:
I see